qt/widgets/instrumentview/src/BankRenderingHelpers.cpp

```diff
@@ -0,0 +1,153 @@
+#include "MantidQtWidgets/InstrumentView/BankRenderingHelpers.h"
+#include "MantidGeometry/Instrument/ComponentInfo.h"
+#include "MantidGeometry/Objects/IObject.h"
+#include "MantidGeometry/Objects/ShapeFactory.h"
+#include "MantidGeometry/Rendering/GeometryHandler.h"
+#include "MantidGeometry/Rendering/OpenGL_Headers.h"
+#include "MantidGeometry/Rendering/ShapeInfo.h"
+#include "MantidKernel/Quat.h"
+
+namespace {
+Mantid::Kernel::Logger g_log("BankRenderingHelpers");
+
+// Round a number up to the nearest power  of 2
+size_t roundToNearestPowerOfTwo(size_t val) {
+  size_t rounded = 2;
+  while (val > rounded)
+    rounded *= 2;
+  return rounded;
+}
+
+void addVertex(Mantid::Kernel::V3D &pos, double xstep, double ystep) {
+  pos += Mantid::Kernel::V3D(xstep * (+0.5), ystep * (-0.5),
+                             0.0); // Adjust to account for the size of a pixel
+  glVertex3f(static_cast<GLfloat>(pos.X()), static_cast<GLfloat>(pos.Y()),
+             static_cast<GLfloat>(pos.Z()));
+}
+
+void extractHexahedron(const Mantid::Geometry::IObject &shape,
+                       std::vector<Mantid::Kernel::V3D> &hex) {
+  const auto &shapeInfo = shape.getGeometryHandler()->shapeInfo();
+  const auto &points = shapeInfo.points();
+  hex.assign(points.begin(), points.begin() + 4);
+}
+
+void rotateHexahedron(std::vector<Mantid::Kernel::V3D> &hex,
+                      const Mantid::Kernel::Quat &rotation) {
+  for (auto &pos : hex)
+    rotation.rotate(pos);
```

Comment:
It would be interesting to see if using `Eigen::Quaterniond` for this gives better performance. i.e. convert from `Kernel::Quat` using `Kernel::EigenConverstionHelpers` once at start of function.

Reply:
May lose on converting all stored hexahedra positions to `Eigen::Vector3D` before applying the transform. See `extractHexahedron` in the same file.